src/ImageRunModal.jsx

```diff
@@ -559,7 +559,8 @@ export class ImageRunModal extends React.Component {
             imageRegistries.push(this.state.searchByRegistry);
         }
 
-        let regexString = searchText;
+        // Filter out characters which are not allowed in RegExp.
+        let regexString = searchText.replace(/[^a-zA-Z0-9_\\.-]/g, "").replaceAll('\\');
```

Comment:
`/g` and replace*All* are redundant, but I suppose it doesn't hurt.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping has a more restricted set of characters to escape. Is this safe against messing up multi-byte unicode? And is it safe to escape any and all characters that way?

Reply:
I don't think it's foolproof, we should probably catch any exception when calling `RegExp()` but I wasn't sure how to handle the exception? Do we just not filter case insensitive?

Reply:
I'm not sure how that is related to case sensitivity. An exception in RegExp() after escaping is an internal programming error. So an Oops isn't actually too bad. Of course you could recover by console.error()'ing the exception and resetting the search form or so, but it's not a much better UX.

Reply:
I think we are approaching this in a very wrong way, we are adding regex to late imo. This is a <Select> sadly but let's see if we can just do form validation.

Also unrelated, we could handle:

![image](https://github.com/cockpit-project/cockpit-podman/assets/67428/0204b1c4-f348-4797-b2c3-ceeeff10c38d)